src/listener.py

```diff
@@ -0,0 +1,190 @@
+def parse_IP(buffer, ethernet_layer):
+	#IP part
+	ip_layer = IPLayer()
+	ip_layer.deserialize(buffer[14:34])
+
+	# Calculate the checksum and verify it.
+	if IPLayer.checksum(buffer[14:34]) == ip_layer.fields["checksum"].get():
+
+		destination_ip = IPAddress.ip2str(ip_layer.fields["dst"].get())
+		source_ip = IPAddress.ip2str(ip_layer.fields["src"].get())
+
+		if not verify_addresses(source_ip, MacAddress.mac2str(ethernet_layer.fields["src"].get())):
+			return
+
+		if destination_ip == MY_ADDRESSES["net1"][1] or destination_ip == MY_ADDRESSES["net2"][1]:
+			pass
+			# The destination is ME!
+			# if ip_layer.fields["protocol"].get()
+
+		# If the packet need to be forward
+		elif destination_ip in ROUTING_TABLE["net1"].keys() or destination_ip in ROUTING_TABLE["net2"].keys():
+
+
+			network = get_network_by_ip(IPAddress.ip2str(destination_ip))
+
+			ethernet_layer.fields["src"].set(MacAddress.str2mac(MY_ADDRESSES[network][0]))
+
+			ethernet_layer.fields["dst"].set(MacAddress.str2mac(ROUTING_TABLE[network][destination_ip]))
+
+			buffer = buffer[14:]
+
+			buffer = ethernet_layer.build() + buffer
+
+			PACKETS_TO_SEND.append((find_network_place(network) , buffer))
+
+			print("Packet forward")
+		else:
+			return
+
+
+
+def parse_ARP(buffer, ethernet_layer):
+	#ARP part
+	arp_layer = ArpLayer()
+	arp_layer.deserialize(buffer[:28])
+
+	destination_ip = IPAddress.ip2str(arp_layer.fields["ip_dst"].get())
+
+	if destination_ip == MY_ADDRESSES["net1"][1] or destination_ip == MY_ADDRESSES["net2"][1]:
+
+		src_ip = IPAddress.ip2str(arp_layer.fields["ip_src"].get())
+		
+		if not verify_addresses(src_ip, MacAddress.mac2str(arp_layer.fields["mac_src"].get())):
+			return
+
+		if arp_layer.fields["opcode"].get() == ArpLayer.OP_WHO_HAS:
+			# ethernet_layer.connect_layer(arp_layer)
+			build_ARP_response(ethernet_layer, arp_layer)
+
+
+def build_ARP_response(ethernet_layer, arp_layer):
+	global PACKETS_TO_SEND
+
+	network = get_network_by_ip(IPAddress.ip2str(arp_layer.fields["ip_src"].get()))
+
+	# Switch the source and destination MAC addresses.
+	ethernet_layer.fields["dst"].set(ethernet_layer.fields["src"].get())
+
+	ethernet_layer.fields["src"].set(MacAddress.str2mac(MY_ADDRESSES[network][0]))
+
+	# Change the addresses in the ARP Layer and the opcode.
+	arp_layer.fields["opcode"].set(ArpLayer.OP_IS_AT)
+
+	arp_layer.fields["ip_dst"].set(arp_layer.fields["ip_src"].get())
+
+	arp_layer.fields["mac_dst"].set(arp_layer.fields["mac_src"].get())
+
+	arp_layer.fields["mac_src"].set(MacAddress.str2mac(MY_ADDRESSES[network][0]))
+
+	arp_layer.fields["ip_src"].set(IPAddress.str2ip(MY_ADDRESSES[network][1]))
+
+	ethernet_layer.connect_layer(arp_layer)
+
+	PACKETS_TO_SEND.append((find_network_place(network) , ethernet_layer.build()))
+
+	print("ARP response sent")
+
+
+
+def create_sockets():
+
+	read_sockets = []
+	read_sockets.append(socket.socket(socket.AF_PACKET, socket.SOCK_RAW, socket.htons(socket.PACKET_OTHERHOST)))
+	read_sockets[0].bind(('net1', 0))
```

Comment:
Maybe make a function that creates and binds a socket for a given interface?
What happens if your router has 36 ports? 🙃